Level 2

ArtemIvlev77 opened the pull request.

ArtemIvlev77 committed:
feat: CRA подготовлена инфраструктура

ArtemIvlev77 committed:
feat: Подготовлена файловая структура

ArtemIvlev77 committed:
feat: сверстаны элементы блока Main: Promo, AboutProject.

ArtemIvlev77 committed:
feat: вёрстка блока main

Добавлены компоненты:
-AboutMe
-Techs
-Portfolio

ArtemIvlev77 committed:
feat: сверстан компонент footer

ArtemIvlev77 committed:
feat: добавлен api

ArtemIvlev77 committed:
feat: добавлены компоненты Prelloader и NotFound

ArtemIvlev77 committed:
feat: добавлен компоненты Form,Register, добавлена валидация формы

ArtemIvlev77 committed:
feat: добавлен компонент ProtectedRoute

ArtemIvlev77 committed:
feat: готово к ревью

ArtemIvlev77 committed:
feat: readme update

coolswood commented:
# Резюме по работе:
### Критерии, влияющие на работоспособность
- [x] <!--10.71--> Проект адаптирован под различные разрешения экрана. Горизонтальный скролл не возникает на разрешениях от 320 пикселей и больше. Нельзя скрывать полосу прокрутки свойством overflow: hidden.
- [ ] <!--10.71--> Проект соответствует сгенерированному под него макету.
`Цвет хедера не соответствует макету. отсутствует кнопка крестик. Имеется лишняя тень у кнопки.`
<img width="406" alt="Снимок экрана 2021-06-25 в 10 36 11" src="https://user-images.githubusercontent.com/25247802/123388798-e1d1b500-d5a1-11eb-8b91-99aeeca521c1.png">
<img width="389" alt="Снимок экрана 2021-06-25 в 10 37 09" src="https://user-images.githubusercontent.com/25247802/123389084-2e1cf500-d5a2-11eb-827a-6884560c2aa6.png">
<img width="70" alt="Снимок экрана 2021-06-25 в 10 37 35" src="https://user-images.githubusercontent.com/25247802/123389130-3d9c3e00-d5a2-11eb-8ce4-887a802347f8.png">

- [ ] <!--10.71--> Все блоки из макета свёрстаны. Корректно работает навигация между страницами и ссылки на внешние ресурсы: ни одна ссылка не ведёт в пустоту или на якорь, внешние ссылки открываются в новой вкладке. На все страницы проекта можно попасть, а скрытые блоки можно отобразить.
`Нет одного из состояний кнопки в карточке (крестик)`
- [ ] <!--10.71--> Отзывчивая вёрстка, которая корректно тянется на всех промежуточных разрешениях.
`Картинка перестраивается неправильно. Кнопка авторизации смещается`
<img width="689" alt="Снимок экрана 2021-06-25 в 10 26 17" src="https://user-images.githubusercontent.com/25247802/123389726-d7fc8180-d5a2-11eb-8870-53ff4a7cfddc.png">
<img width="1236" alt="Снимок экрана 2021-06-25 в 10 37 57" src="https://user-images.githubusercontent.com/25247802/123389780-e77bca80-d5a2-11eb-9cbc-648d8bc6e394.png">

- [x] <!--2.86--> Отсутствуют ошибки валидации при сборке приложения или в валидаторе.
- [x] <!--2.86--> В коде используется семантическая разметка: применяются семантические теги, выбор элементов при вёрстке корректен (параграф должен быть параграфом, список — списком); структура DOM-дерева состоит не только из контейнеров div.
- [x] <!--2.86--> Для позиционирования элементов выбран верный подход, описанный корректным синтаксисом.
- [x] <!--2.86--> Каркас макета реализован на Flex layout или Grid layout;
- [x] <!--2.86--> Инфраструктурные файлы проекта созданы через CRA.
- [x] <!--2.86--> Работа модальных окон настроена.
- [x] <!--2.86--> Разметка портирована в JSX:
- [x] <!--2.86--> разметка заключена в ( ), лишние пустые <div> для этого не используются;
- [x] <!--2.86--> вынесена в соответствующие ей компоненты.
- [x] <!--2.86--> В проекте есть:
- [x] <!--2.86--> директория images/ с картинками,
- [x] <!--2.86--> директория components/ с JS и CSS-файлами компонентов,
- [x] <!--2.86--> директория fonts/ со шрифтами.

### Хорошие практики
- [x] <!--2.14--> Именование классов сделано по БЭМ.
- [x] <!--2.14--> Кнопки, инпуты и ссылки реализованы во всех состояниях, указанных в макете.
- [x] <!--2.14--> В макете есть одинаковые элементы. Для их оформления должен быть переиспользован один компонент.
- [x] <!--2.14--> Шрифты подключены через @font-face.
- [x] <!--2.14--> reset.css запрещен;
- [x] <!--2.14--> Элементы формы должны выделяться, когда на них установлен фокус. Состояние фокуса, а также outline стилизованы в соответствии с макетом или на своё усмотрение.
- [x] <!--2.14--> У формы должны быть заданы плейсхолдеры, валидированы обязательные поля.

### Рекомендации
- [x] <!--1.67--> Для изображений задан атрибут alt с подходящий значением.
- [x] <!--1.67--> Для каждого шрифта указаны альтернативные варианты из системных шрифтов.
- [x] <!--1.67--> Растровые и векторные изображения оптимизированы;

## Количество баллов: 68

ArtemIvlev77 committed:
feat: Исправлено согласно комментариям ревью:

1.Цвет хедера не соответствует макету.  Имеется лишняя тень у кнопки.
2. Rартинка перестраивается неправильно. Кнопка авторизации смещается

ArtemIvlev77 committed:
feat: исправлены состояния кнопки карточки

coolswood commented:
# Резюме по работе:
### Критерии, влияющие на работоспособность
- [x] <!--10.71--> Проект адаптирован под различные разрешения экрана. Горизонтальный скролл не возникает на разрешениях от 320 пикселей и больше. Нельзя скрывать полосу прокрутки свойством overflow: hidden.
- [ ] <!--10.71--> Проект соответствует сгенерированному под него макету.
`Отсутствует кнопка "короткометражка" и кнопки в хедере.`
<img width="289" alt="Снимок экрана 2021-06-29 в 17 29 00" src="https://user-images.githubusercontent.com/25247802/123816633-23df5b80-d900-11eb-917e-08d5197c1dc7.png">
<img width="971" alt="Снимок экрана 2021-06-29 в 17 29 52" src="https://user-images.githubusercontent.com/25247802/123816721-36f22b80-d900-11eb-89f2-ac963f76a924.png">

- [x] <!--10.71--> Все блоки из макета свёрстаны. Корректно работает навигация между страницами и ссылки на внешние ресурсы: ни одна ссылка не ведёт в пустоту или на якорь, внешние ссылки открываются в новой вкладке. На все страницы проекта можно попасть, а скрытые блоки можно отобразить.
`Нет одного из состояний кнопки в карточке (крестик)`
- [ ] <!--10.71--> Отзывчивая вёрстка, которая корректно тянется на всех промежуточных разрешениях.
`Картинка перестраивается неправильно.`
<img width="689" alt="Снимок экрана 2021-06-25 в 10 26 17" src="https://user-images.githubusercontent.com/25247802/123389726-d7fc8180-d5a2-11eb-8870-53ff4a7cfddc.png">

- [x] <!--2.86--> Отсутствуют ошибки валидации при сборке приложения или в валидаторе.
- [x] <!--2.86--> В коде используется семантическая разметка: применяются семантические теги, выбор элементов при вёрстке корректен (параграф должен быть параграфом, список — списком); структура DOM-дерева состоит не только из контейнеров div.
- [x] <!--2.86--> Для позиционирования элементов выбран верный подход, описанный корректным синтаксисом.
- [x] <!--2.86--> Каркас макета реализован на Flex layout или Grid layout;
- [x] <!--2.86--> Инфраструктурные файлы проекта созданы через CRA.
- [x] <!--2.86--> Работа модальных окон настроена.
- [x] <!--2.86--> Разметка портирована в JSX:
- [x] <!--2.86--> разметка заключена в ( ), лишние пустые <div> для этого не используются;
- [x] <!--2.86--> вынесена в соответствующие ей компоненты.
- [x] <!--2.86--> В проекте есть:
- [x] <!--2.86--> директория images/ с картинками,
- [x] <!--2.86--> директория components/ с JS и CSS-файлами компонентов,
- [x] <!--2.86--> директория fonts/ со шрифтами.

### Хорошие практики
- [x] <!--2.14--> Именование классов сделано по БЭМ.
- [x] <!--2.14--> Кнопки, инпуты и ссылки реализованы во всех состояниях, указанных в макете.
- [x] <!--2.14--> В макете есть одинаковые элементы. Для их оформления должен быть переиспользован один компонент.
- [x] <!--2.14--> Шрифты подключены через @font-face.
- [x] <!--2.14--> reset.css запрещен;
- [x] <!--2.14--> Элементы формы должны выделяться, когда на них установлен фокус. Состояние фокуса, а также outline стилизованы в соответствии с макетом или на своё усмотрение.
- [x] <!--2.14--> У формы должны быть заданы плейсхолдеры, валидированы обязательные поля.

### Рекомендации
- [x] <!--1.67--> Для изображений задан атрибут alt с подходящий значением.
- [x] <!--1.67--> Для каждого шрифта указаны альтернативные варианты из системных шрифтов.
- [x] <!--1.67--> Растровые и векторные изображения оптимизированы;

## Количество баллов: 79

ArtemIvlev77 committed:
fix: замечания  ревью

ArtemIvlev77 commented:
> # Резюме по работе:
> ### Критерии, влияющие на работоспособность
> * [x]   Проект адаптирован под различные разрешения экрана. Горизонтальный скролл не возникает на разрешениях от 320 пикселей и больше. Нельзя скрывать полосу прокрутки свойством overflow: hidden.
> * [ ]   Проект соответствует сгенерированному под него макету.
>   `Отсутствует кнопка "короткометражка" и кнопки в хедере.`
> 
> <img alt="Снимок экрана 2021-06-29 в 17 29 00" width="289" src="https://user-images.githubusercontent.com/25247802/123816633-23df5b80-d900-11eb-917e-08d5197c1dc7.png">
> 
> <img alt="Снимок экрана 2021-06-29 в 17 29 52" width="971" src="https://user-images.githubusercontent.com/25247802/123816721-36f22b80-d900-11eb-89f2-ac963f76a924.png">
> 
> * [x]   Все блоки из макета свёрстаны. Корректно работает навигация между страницами и ссылки на внешние ресурсы: ни одна ссылка не ведёт в пустоту или на якорь, внешние ссылки открываются в новой вкладке. На все страницы проекта можно попасть, а скрытые блоки можно отобразить.
>   `Нет одного из состояний кнопки в карточке (крестик)`
> * [ ]   Отзывчивая вёрстка, которая корректно тянется на всех промежуточных разрешениях.
>   `Картинка перестраивается неправильно.`
> 
> <img alt="Снимок экрана 2021-06-25 в 10 26 17" width="689" src="https://user-images.githubusercontent.com/25247802/123389726-d7fc8180-d5a2-11eb-8870-53ff4a7cfddc.png">
> 
> * [x]   Отсутствуют ошибки валидации при сборке приложения или в валидаторе.
> * [x]   В коде используется семантическая разметка: применяются семантические теги, выбор элементов при вёрстке корректен (параграф должен быть параграфом, список — списком); структура DOM-дерева состоит не только из контейнеров div.
> * [x]   Для позиционирования элементов выбран верный подход, описанный корректным синтаксисом.
> * [x]   Каркас макета реализован на Flex layout или Grid layout;
> * [x]   Инфраструктурные файлы проекта созданы через CRA.
> * [x]   Работа модальных окон настроена.
> * [x]   Разметка портирована в JSX:
> * [x]   разметка заключена в ( ), лишние пустые  для этого не используются;
> * [x]   вынесена в соответствующие ей компоненты.
> * [x]   В проекте есть:
> * [x]   директория images/ с картинками,
> * [x]   директория components/ с JS и CSS-файлами компонентов,
> * [x]   директория fonts/ со шрифтами.
> 
> ### Хорошие практики
> * [x]   Именование классов сделано по БЭМ.
> * [x]   Кнопки, инпуты и ссылки реализованы во всех состояниях, указанных в макете.
> * [x]   В макете есть одинаковые элементы. Для их оформления должен быть переиспользован один компонент.
> * [x]   Шрифты подключены через @font-face.
> * [x]   reset.css запрещен;
> * [x]   Элементы формы должны выделяться, когда на них установлен фокус. Состояние фокуса, а также outline стилизованы в соответствии с макетом или на своё усмотрение.
> * [x]   У формы должны быть заданы плейсхолдеры, валидированы обязательные поля.
> 
> ### Рекомендации
> * [x]   Для изображений задан атрибут alt с подходящий значением.
> * [x]   Для каждого шрифта указаны альтернативные варианты из системных шрифтов.
> * [x]   Растровые и векторные изображения оптимизированы;
> 
> ## Количество баллов: 79

При построение элементов с помощью flex-box, используется свойство order, изменяющее порядок построения элементов в зависимости значения media-screen. После изменения значения и обновления страницы элементы выстраиваются согласно макету.

coolswood commented:
# Резюме по работе:
### Критерии, влияющие на работоспособность
- [x] <!--10.71--> Проект адаптирован под различные разрешения экрана. Горизонтальный скролл не возникает на разрешениях от 320 пикселей и больше. Нельзя скрывать полосу прокрутки свойством overflow: hidden.
- [ ] <!--10.71--> Проект соответствует сгенерированному под него макету.
`Отсутствует кнопка крестик в бургерном меню`
<img width="375" alt="Снимок экрана 2021-07-01 в 12 53 39" src="https://user-images.githubusercontent.com/25247802/124106116-966d4a00-da6c-11eb-88d3-3dd4f1c19988.png">


- [x] <!--10.71--> Все блоки из макета свёрстаны. Корректно работает навигация между страницами и ссылки на внешние ресурсы: ни одна ссылка не ведёт в пустоту или на якорь, внешние ссылки открываются в новой вкладке. На все страницы проекта можно попасть, а скрытые блоки можно отобразить.
`Нет одного из состояний кнопки в карточке (крестик)`
- [ ] <!--10.71--> Отзывчивая вёрстка, которая корректно тянется на всех промежуточных разрешениях.
`при изменении ширины браузера картинка оказывается под текстом. Через некоторое время она перемещается наверх, видимо проблема в некорректной анимации. При изменении ширины браузера измерения сразу должны быть правильными`

<img width="689" alt="Снимок экрана 2021-06-25 в 10 26 17" src="https://user-images.githubusercontent.com/25247802/123389726-d7fc8180-d5a2-11eb-8870-53ff4a7cfddc.png">

- [x] <!--2.86--> Отсутствуют ошибки валидации при сборке приложения или в валидаторе.
- [x] <!--2.86--> В коде используется семантическая разметка: применяются семантические теги, выбор элементов при вёрстке корректен (параграф должен быть параграфом, список — списком); структура DOM-дерева состоит не только из контейнеров div.
- [x] <!--2.86--> Для позиционирования элементов выбран верный подход, описанный корректным синтаксисом.
- [x] <!--2.86--> Каркас макета реализован на Flex layout или Grid layout;
- [x] <!--2.86--> Инфраструктурные файлы проекта созданы через CRA.
- [x] <!--2.86--> Работа модальных окон настроена.
- [x] <!--2.86--> Разметка портирована в JSX:
- [x] <!--2.86--> разметка заключена в ( ), лишние пустые <div> для этого не используются;
- [x] <!--2.86--> вынесена в соответствующие ей компоненты.
- [x] <!--2.86--> В проекте есть:
- [x] <!--2.86--> директория images/ с картинками,
- [x] <!--2.86--> директория components/ с JS и CSS-файлами компонентов,
- [x] <!--2.86--> директория fonts/ со шрифтами.

### Хорошие практики
- [x] <!--2.14--> Именование классов сделано по БЭМ.
- [x] <!--2.14--> Кнопки, инпуты и ссылки реализованы во всех состояниях, указанных в макете.
- [x] <!--2.14--> В макете есть одинаковые элементы. Для их оформления должен быть переиспользован один компонент.
- [x] <!--2.14--> Шрифты подключены через @font-face.
- [x] <!--2.14--> reset.css запрещен;
- [x] <!--2.14--> Элементы формы должны выделяться, когда на них установлен фокус. Состояние фокуса, а также outline стилизованы в соответствии с макетом или на своё усмотрение.
- [x] <!--2.14--> У формы должны быть заданы плейсхолдеры, валидированы обязательные поля.

### Рекомендации
- [x] <!--1.67--> Для изображений задан атрибут alt с подходящий значением.
- [x] <!--1.67--> Для каждого шрифта указаны альтернативные варианты из системных шрифтов.
- [x] <!--1.67--> Растровые и векторные изображения оптимизированы;

## Количество баллов: 79

ArtemIvlev77 committed:
fix: исправлена кнопка бургер меню  и корректное отображение элементов при изменении ширины экрана

ArtemIvlev77 committed:
fix: исправлено расстояние в header

coolswood commented:
# Резюме по работе:
### Критерии, влияющие на работоспособность
- [x] <!--10.71--> Проект адаптирован под различные разрешения экрана. Горизонтальный скролл не возникает на разрешениях от 320 пикселей и больше. Нельзя скрывать полосу прокрутки свойством overflow: hidden.
- [x] <!--10.71--> Проект соответствует сгенерированному под него макету.
`Отсутствует кнопка крестик в бургерном меню`
<img width="375" alt="Снимок экрана 2021-07-01 в 12 53 39" src="https://user-images.githubusercontent.com/25247802/124106116-966d4a00-da6c-11eb-88d3-3dd4f1c19988.png">


- [x] <!--10.71--> Все блоки из макета свёрстаны. Корректно работает навигация между страницами и ссылки на внешние ресурсы: ни одна ссылка не ведёт в пустоту или на якорь, внешние ссылки открываются в новой вкладке. На все страницы проекта можно попасть, а скрытые блоки можно отобразить.
`Нет одного из состояний кнопки в карточке (крестик)`
- [x] <!--10.71--> Отзывчивая вёрстка, которая корректно тянется на всех промежуточных разрешениях.
`при изменении ширины браузера картинка оказывается под текстом. Через некоторое время она перемещается наверх, видимо проблема в некорректной анимации. При изменении ширины браузера измерения сразу должны быть правильными`

<img width="689" alt="Снимок экрана 2021-06-25 в 10 26 17" src="https://user-images.githubusercontent.com/25247802/123389726-d7fc8180-d5a2-11eb-8870-53ff4a7cfddc.png">

- [x] <!--2.86--> Отсутствуют ошибки валидации при сборке приложения или в валидаторе.
- [x] <!--2.86--> В коде используется семантическая разметка: применяются семантические теги, выбор элементов при вёрстке корректен (параграф должен быть параграфом, список — списком); структура DOM-дерева состоит не только из контейнеров div.
- [x] <!--2.86--> Для позиционирования элементов выбран верный подход, описанный корректным синтаксисом.
- [x] <!--2.86--> Каркас макета реализован на Flex layout или Grid layout;
- [x] <!--2.86--> Инфраструктурные файлы проекта созданы через CRA.
- [x] <!--2.86--> Работа модальных окон настроена.
- [x] <!--2.86--> Разметка портирована в JSX:
- [x] <!--2.86--> разметка заключена в ( ), лишние пустые <div> для этого не используются;
- [x] <!--2.86--> вынесена в соответствующие ей компоненты.
- [x] <!--2.86--> В проекте есть:
- [x] <!--2.86--> директория images/ с картинками,
- [x] <!--2.86--> директория components/ с JS и CSS-файлами компонентов,
- [x] <!--2.86--> директория fonts/ со шрифтами.

### Хорошие практики
- [x] <!--2.14--> Именование классов сделано по БЭМ.
- [x] <!--2.14--> Кнопки, инпуты и ссылки реализованы во всех состояниях, указанных в макете.
- [x] <!--2.14--> В макете есть одинаковые элементы. Для их оформления должен быть переиспользован один компонент.
- [x] <!--2.14--> Шрифты подключены через @font-face.
- [x] <!--2.14--> reset.css запрещен;
- [x] <!--2.14--> Элементы формы должны выделяться, когда на них установлен фокус. Состояние фокуса, а также outline стилизованы в соответствии с макетом или на своё усмотрение.
- [x] <!--2.14--> У формы должны быть заданы плейсхолдеры, валидированы обязательные поля.

### Рекомендации
- [x] <!--1.67--> Для изображений задан атрибут alt с подходящий значением.
- [x] <!--1.67--> Для каждого шрифта указаны альтернативные варианты из системных шрифтов.
- [x] <!--1.67--> Растровые и векторные изображения оптимизированы;

## Количество баллов: 100
